### PR TITLE
Support extensions with more than 1 dot.

### PR DESCRIPTION
**Pull request by leoschwarz**
Currenty file type detection based on extension fails, if there is a
filename of the form `mesh.1.off` instead of `mesh.off` (with only a
single dot). This fix replaces the custom code with the default
`os.path.splitext` function which always only returns the rightmost part
or an empty string.

### CODE REVIEW

File: test/test_helpers.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
     'test.vtk',
     'test.vtu',
     'test.xmf',
+    'test.1.off',
```
Comment (leoschwarz) [inline test/test_helpers.py]: I'm not sure if you want to merge it like this or if I should add an example for all variants or a dedicated test?

**Comment by nschloe** [comment]
This won't play nicely with double extensions like `.mesh.gz` (see the PERMAS format). Not sure why we don't have a test for this.

**Comment by leoschwarz** [comment]
My bad, I should have slept over the commit before sending it and do more testing, thanks for pointing this out. I can add a test for that tomorrow.

I guess another and more robust approach would be to split at the first dot and check if it is a known extension (`.mesh.gz` would be caught this way) and if not part by part of it would be stripped away from left to right until something is found or nothing is left (`.1.mesh` would not resolve, but then `.mesh` would).

**Comment by gdmcbain** [comment]
On double extensions, [above](https://github.com/nschloe/meshio/pull/228#issuecomment-387528059)

> This won't play nicely with double extensions like .mesh.gz (see the PERMAS format). Not sure why we don't have a test for this.

I thought there was.  I had mentioned there being a 'test.post.gz' in test/test_helpers.py in the discussion of #200 on [2018-02-08](https://github.com/nschloe/meshio/issues/200#issuecomment-364001420) but see that the second suffix was lopped in 6ea9fd9.  I'm not immediately sure why it was lopped.

**Comment by leoschwarz** [comment]
@gdmcbain Actually when I try adding the tests that were removed I get the error `OSError: Not a gzipped file (b'!\n')`.

I think this should not be merged before the there are some tests for it.

**Comment by nschloe** [comment]
`test_helpers.py` got kicked out in another PR. Can you just move the new file names into the respective `test_*` files?

**Pull request merged**